examples/darwin-framework-tool/commands/common/DFTKeypair.mm

```diff
@@ -0,0 +1,239 @@
+/*
+ *   Copyright (c) 2024 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#import "DFTKeypair.h"
+
+#import "CHIPCommandStorageDelegate.h"
+#import "ControllerStorage.h"
+
+#define CAKeyTag "com.apple.matter.commissioner.ca.issuer.id"
+#define KeySize "256"
+#define CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES 16
+
+static NSString * const kKeychainLabel = @"Chip Tool Keypair";
+static NSString * const kOperationalCredentialsIssuerKeypairStorage = @"ChipToolOpCredsCAKey";
+static NSString * const kOperationalCredentialsIPK = @"ChipToolOpCredsIPK";
+
+// Error Descriptions
+NSString * const kErrorDomain = @"Error";
+NSString * const kErrorFailedToStoreKeypairData = @"Failed to store keypair data in storage.";
+NSString * const kErrorInvalidIPKData = @"Invalid IPK data.";
+NSString * const kErrorFailedToGenerateIPK = @"Failed to generate random bytes for IPK.";
+NSString * const kErrorFailedToStoreIPK = @"Failed to store IPK in storage.";
+NSString * const kPublicKeyRetrievalFailureReason = @"Failed to retrieve the public key from the private key. This may occur if the private key was created without storing the corresponding public key in the keychain, or if the system cannot reconstruct the public key.";
+
+@interface DFTKeypair ()
+@property (readonly) SecKeyRef privateKey;
+@property (readonly) SecKeyRef publicKey;
+@property (readonly) NSData * ipk;
+@end
+
+@implementation DFTKeypair
++ (instancetype)createKeypairWithStorage:(id)storage error:(NSError * _Nullable __autoreleasing *)error
+{
+    __auto_type * keypair = [[self alloc] init];
+    if (![keypair setupKeys:storage error:error] || ![keypair setupIPK:storage error:error]) {
+        return nil;
+    }
+
+    return keypair;
+}
+
+- (NSData *)signMessageECDSA_DER:(NSData *)message
+{
+    if (!_privateKey) {
+        NSLog(@"Error: Private key is not available for signing.");
+        return nil;
+    }
+
+    CFErrorRef cfError = nil;
+    CFDataRef signatureData = SecKeyCreateSignature(_privateKey,
+        kSecKeyAlgorithmECDSASignatureMessageX962SHA256,
+        (__bridge CFDataRef) message,
+        &cfError);
+
+    if (!signatureData) {
+        NSError * error = (__bridge_transfer NSError *) cfError;
```

Comment:
Hmm, should the corresponding cast in MTRTestKeys also be a __bridge_transfer?

And in `CHIPTool/Framework Helpers/FabricKeys.m`?

Followup as needed.